fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
         log.exiting(CLASSNAME, METHODNAME);
         return queryData;
     }
+
+    /**
+     * Contains special logic for handling reverse chained reference search parameters.
+     * <p>
+     * A select statement is built to realize the reverse chaining logic required. Here is a sample
+     * reverse chained query for a Patient given this search parameter: _has:Observation:patient:code=1234
+     *
+     * <pre>
+     * SELECT
+     *   CLR0.LOGICAL_ID
+     * FROM
+     *   Patient_LOGICAL_RESOURCES AS CLR0
+     *   JOIN Patient_RESOURCES AS CR0 ON CR0.RESOURCE_ID = CLR0.CURRENT_RESOURCE_ID AND CR0.IS_DELETED = 'N'
+     * WHERE
+     *   EXISTS (
+     *     SELECT
+     *       1
+     *     FROM
+     *       Observation_TOKEN_VALUES_V AS CP1
+     *       JOIN Observation_LOGICAL_RESOURCES AS CLR1 ON CLR1.LOGICAL_RESOURCE_ID = CP1.LOGICAL_RESOURCE_ID
+     *       JOIN Observation_RESOURCES AS CR1 ON CR1.RESOURCE_ID = CLR1.CURRENT_RESOURCE_ID AND CR1.IS_DELETED = 'N'
+     *       JOIN Observation_TOKEN_VALUES_V AS CP2 ON CP2.LOGICAL_RESOURCE_ID = CLR1.LOGICAL_RESOURCE_ID
+     *       AND (
+     *         CP2.PARAMETER_NAME_ID = 1073
+     *         AND ((CP2.TOKEN_VALUE = ?))
+     *       )
+     *     WHERE
+     *       CP1.TOKEN_VALUE = CLR0.LOGICAL_ID
+     *       AND CP1.PARAMETER_NAME_ID = 1274
+     *       AND CP1.CODE_SYSTEM_ID = 20004
+     *   )
+     * </pre>
+     *
+     * @see https://www.hl7.org/fhir/search.html#has
+     * @param resourceType
+     *                  - The resource type being searched.
+     * @param queryParm
+     *                  - A Parameter representing a reverse chained query.
+     * @return SqlQueryData
+     *                  - The query segment for a reverse chained parameter reference search.
+     * @throws Exception
+     */
+    @Override
+    protected SqlQueryData processReverseChainedReferenceParm(Class<?> resourceType, QueryParameter queryParm) throws Exception {
+        final String METHODNAME = "processReverseChainedReferenceParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+
+        final String CR = "CR";
+        final String CLR = "CLR";
+        final String CP = "CP";
```

Comment:
I'm almost positive these are repeated constants, maybe they should be top level constants for consistency?

Reply:
moved out of this method and processChainedReferenceParm() and made them global